src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java

```diff
@@ -1125,6 +1125,28 @@ public void testDeleteEquipment() {
                 MatcherEquipmentDeletionInfos.createMatcherEquipmentDeletionInfos(ModificationType.EQUIPMENT_DELETION, "v1lcc", "HVDC_CONVERTER_STATION", Set.of("s1")));
 
         testNetworkModificationsCount(TEST_GROUP_ID, 13);
+
+        // delete voltage level
+        webTestClient.delete().uri(uriString, TEST_NETWORK_ID, "VOLTAGE_LEVEL", "v5")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(EquipmentDeletionInfos.class)
+                .value(modifications -> modifications.get(0),
+                        MatcherEquipmentDeletionInfos.createMatcherEquipmentDeletionInfos(ModificationType.EQUIPMENT_DELETION, "v5", "VOLTAGE_LEVEL", Set.of("s3")));
+
+        testNetworkModificationsCount(TEST_GROUP_ID, 14);
+
+        // delete substation
+        webTestClient.delete().uri(uriString, TEST_NETWORK_ID, "SUBSTATION", "s3")
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(EquipmentDeletionInfos.class)
+                .value(modifications -> modifications.get(0),
+                        MatcherEquipmentDeletionInfos.createMatcherEquipmentDeletionInfos(ModificationType.EQUIPMENT_DELETION, "s3", "SUBSTATION", Set.of()));
+
+        testNetworkModificationsCount(TEST_GROUP_ID, 15);
     }
```

Comment:
Maybe test also failing voltage level and substation deletion (when lines or transformers are present)
Test also that all equipments in the voltage level are properly deleted (injections ...)

Reply:
I added tests when deletion fails and some check that the substation or the vl is deleted. 
For injections that are deleted, this behavior is tested in NetworkStoreIt